resolwe_bio/processes/chip_seq/macs2.yml

```diff
@@ -481,132 +748,133 @@
+        re-save-file control_lambda "${CASE_NAME}_control_lambda.bdg"
+        re-save-file control_lambda_bigwig "${CASE_NAME}_control_lambda.bw"
       {% endif %}
 
       # Get post-peakcall QC metrics
       {% if not settings.broad %}
         # Sort narrowPeak by p values
-        LC_COLLATE=C sort -k 8gr,8gr "${NAME}_peaks.narrowPeak" | \
-        awk 'BEGIN{OFS="\t"}{$4="Peak_"NR; if ($2<0) $2=0; if ($3<0) $3=0; print $0}' > "${NAME}_peaks_tmp.narrowPeak"
+        LC_COLLATE=C sort -k 8gr,8gr "${CASE_NAME}_peaks.narrowPeak" | \
+        awk 'BEGIN{OFS="\t"}{$4="Peak_"NR; if ($2<0) $2=0; if ($3<0) $3=0; print $0}' > "${CASE_NAME}_peaks_tmp.narrowPeak"
 
         # Cap number of peaks
-        head -n {{ settings.cap_num or -0 }} "${NAME}_peaks_tmp.narrowPeak" > "${NAME}_peaks.narrowPeak"
+        head -n {{ settings.cap_num or -0 }} "${CASE_NAME}_peaks_tmp.narrowPeak" > "${CASE_NAME}_peaks.narrowPeak"
 
-        {% if case_prepeak %}
-          cp {{ case_prepeak.tagalign.file }} "${NAME}.tagAlign"
+        # Use either filtered tagAlign file or compute regular tagAlign file for the case sample
+        {% if tagalign %}
+          TAGALIGN="${CASE_TAGALIGN}"
         {% else %}
+          TAGALIGN="${CASE_NAME}_nonfiltered.tagAlign"
           # Transform case bam file to tagAlign for either single-end or paired-end reads
           if [ $(samtools view -c -f 1 {{ case.bam.file }}) -eq 0 ]; then
             bedtools bamtobed -i {{ case.bam.file }} | \
-            awk 'BEGIN{OFS="\t"}{$4="N";$5="1000";print $0}' > "${NAME}.tagAlign"
+            awk 'BEGIN{OFS="\t"}{$4="N";$5="1000";print $0}' > "${TAGALIGN}"
             re-checkrc "Conversion to tagAlign failed"
           else
-            samtools sort -n {{ case.bam.file }} "${NAME}_sorted"
-            samtools fixmate -r "${NAME}_sorted.bam" "${NAME}.bam"
-            samtools view -b -f 2 "${NAME}.bam" | \
+            samtools sort -n {{ case.bam.file }} "${CASE_NAME}_sorted"
+            samtools fixmate -r "${CASE_NAME}_sorted.bam" "${CASE_NAME}.bam"
+            samtools view -b -f 2 "${CASE_NAME}.bam" | \
             bedtools bamtobed -bedpe -mate1 -i - | \
-            awk 'BEGIN{OFS="\t"}{printf "%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n",$1,$2,$3,$9,$4,$5,$6,$10}' > "${NAME}.tagAlign"
+            awk 'BEGIN{OFS="\t"}{printf "%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n",$1,$2,$3,$9,$4,$5,$6,$10}' > "${TAGALIGN}"
             re-checkrc "Conversion to tagAlign failed"
           fi
         {% endif %}
 
-        # Get fragment length
+        # Use user-defined fragment length if extsize parameter is set
         {% if settings.extsize %}
           FRAGLEN={{ settings.extsize }}
-        {% elif case_prepeak %}
-          FRAGLEN={{ case_prepeak.fraglen }}
-        {% else %}
-          FRAGLEN=$(macs2 predictd -i {{ case.bam.file }} 2>&1 >/dev/null | grep -e 'predicted fragment length' | rev | cut -d ' ' -f 3 | rev)
```

Comment:
There might be a case where ǹeither `extsize` or `tagalign` are defined. In this case we need this line to estimate fragment length from the bam file. It is later used to shift the reads in https://github.com/genialis/resolwe-bio/pull/695/files#diff-d94f03937f375b4a82ac634d68ad0faaR854.

Reply:
Prepeak QC metrics are always computed (even if the filtered tagalign file is not to be used for the macs analysis), so fragment length is always extracted for the `case` input: https://github.com/genialis/resolwe-bio/pull/695/files#diff-d94f03937f375b4a82ac634d68ad0faaR669